### ISSUE TITLE
regression: delete by destination broken for tables with owned vindexes

### ISSUE THREAD

**Issue by sougou** [vitessio/vitess#6541]
Create a table with an owned vindex, and issue a delete by destination as follows:

```
mysql> delete from `user[-]`.user;
ERROR 1105 (HY000): vtgate: http://sougou-lap1:12345/: deleteVindexEntries: target: user.80-.master, used tablet: test-4 (sougou-lap1): vttablet: rpc error: code = InvalidArgument desc = syntax error at position 9 (CallerID: userData1)
```

The error is because of vtgate sending an incorrect select to vttablet:
```
Execute sougou@127.0.0.1:41480(Mysql)   sougou  'userData1'     'sougou'        2020-08-05 12:50:39.219541      2020-08-05 12:50:39.219884      0.000343                "select , name from user for update"    map[]   0       ""      none    0.000000        0.000000        0       0       "syntax error at position 9 (CallerID: userData1)"
```

Previously, we were not deleting owned rows for `ByDestination` plans, but a recent change started using that code path, but the planbuilder wasn't building the select query properly.

**Comment by sougou** [comment]
The fix will need to be backported to 7.0. A different temp fix is being deployed for 6.0.

**Comment by rafael** [comment]
For 6.0, we just reverted to the old functionality: https://github.com/vitessio/vitess/pull/6542

**Comment by harshit-gangal** [comment]
The same issue will happen for update as well.

**Comment by harshit-gangal** [comment]
Closed by #6554
